src/sedcli_main.c

```diff
@@ -424,64 +424,76 @@ static void print_sed_status(int status)
+	sedcli_printf(LOG_INFO, "\tMBR Enabled       : %s\n", discv->sed_lvl0_discv.sed_locking.mbr_en ? "Y" : "N");
+	sedcli_printf(LOG_INFO, "\tMBR Done          : %s\n", discv->sed_lvl0_discv.sed_locking.mbr_done ? "Y" : "N");
+
+	/* Printing Geometry Supported Features*/
+	sedcli_printf(LOG_INFO, "\nSED GEOMETRY FEATURES SUPPORTED\n");
+	sedcli_printf(LOG_INFO, "\tAlignment required    : %s\n", discv->sed_lvl0_discv.sed_geo.rsvd_align.align ? "Y" : "N");
+	sedcli_printf(LOG_INFO, "\tLogical Block Size    : %d\n", be32toh(discv->sed_lvl0_discv.sed_geo.logical_blk_sz));
+	sedcli_printf(LOG_INFO, "\tAlignment Granularity : %ld\n", be64toh(discv->sed_lvl0_discv.sed_geo.alignmnt_granlrty));
+	sedcli_printf(LOG_INFO, "\tLowest Aligned LBA    : %ld\n",  be64toh(discv->sed_lvl0_discv.sed_geo.lowest_aligned_lba));
+
+	/* Printing Datastore Fetaures */
+	sedcli_printf(LOG_INFO, "\nSED DATASTORE FEATURES SUPPORTED\n");
+	sedcli_printf(LOG_INFO, "\tMax DataStore tables       : %d\n", be16toh(discv->sed_lvl0_discv.sed_datastr.max_num_datastr_tbls));
+	sedcli_printf(LOG_INFO, "\tMax size DataStore tables  : %d\n", be32toh(discv->sed_lvl0_discv.sed_datastr.max_total_size_datstr_tbls));
+	sedcli_printf(LOG_INFO, "\tDataStore table size align : %d\n", be32toh(discv->sed_lvl0_discv.sed_datastr.datastr_tbl_size_align));
+
+	/* Printing Opalv100 Features */
+	if (!be16toh(discv->sed_lvl0_discv.sed_opalv100.v1_base_comid))
+		sedcli_printf(LOG_INFO, "\nSED Opal v1.00 FEATURES NOT SUPPORTED\n");
+	else {
+		sedcli_printf(LOG_INFO, "\nSED Opal v1.00 FEATURES SUPPORTED\n");
+		sedcli_printf(LOG_INFO, "\tBase ComID       : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv100.v1_base_comid));
+		sedcli_printf(LOG_INFO, "\tNumber of ComIDs : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv100.v1_comid_num));
+	}
+
+	/* Printing Opalv200 Features */
+	sedcli_printf(LOG_INFO, "\nSED Opal v2.00 FEATURES SUPPORTED\n");
+	sedcli_printf(LOG_INFO, "\tBase ComID                      : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv200.base_comid));
+	sedcli_printf(LOG_INFO, "\tNumber of ComIDs                : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv200.comid_num));
+	sedcli_printf(LOG_INFO, "\tRange Crossing Behavior         : %d\n", discv->sed_lvl0_discv.sed_opalv200.rangecross_rsvd.range_crossing ? 0 : 1);
+	sedcli_printf(LOG_INFO, "\tAdmin Authorities LSP Supported : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv200.admin_lp_auth_num));
+	sedcli_printf(LOG_INFO, "\tUser Authorities LSP Supported  : %d\n", be16toh(discv->sed_lvl0_discv.sed_opalv200.user_lp_auth_num));
+	sedcli_printf(LOG_INFO, "\tInitial PIN                     : %d\n", discv->sed_lvl0_discv.sed_opalv200.init_pin);
+	sedcli_printf(LOG_INFO, "\tRevert PIN                      : %d\n", discv->sed_lvl0_discv.sed_opalv200.revert_pin);
+
+	/* Printing TPer Properties */
+	sedcli_printf(LOG_INFO, "\nTPER PROPERTIES\n");
+	for (int i = 0; i < 23; i++) {
```

Comment:
instead of "23" either define or array_size should be used

Reply:
will define the value in libsed and use it here